rfcs/0006-alternative-suspend-control/README.md

```diff
@@ -0,0 +1,181 @@
+
+### Non-Goals
+
+The deprecation plan for the `.spec.suspend` field is out of scope for this RFC.
+
+
+## Proposal
+
+Register a resource object status for the suspended state. The flux object status `suspended` holds the current state of suspension from reconciliation for the resource.
+
+Register a flux resource metadata key with a temporal suspend semantic with suspend status handling implemented in each of the controllers and in the flux cli.  The annotation `reconcile.fluxcd.io/suspendedDuring` defines when to suspend a resource through a period of time.  Supports a cron format including tags and ranges.  It's presence is used by flux controllers to evaluate and set the suspend status of the resource.  Is removed by a resume command.
+
+### User Stories
+
+#### Suspend/Resume without Generation Roll
+
+Currently when a resource is set to suspended or resumed the `.spec.suspend` field is mutated which results in a roll of the generation number in the `.metadata.generation` and fields `.status.observedGeneration` number.  The community believes that the generation change for this reason is not in alignment with gitops principles.
+
+The flux controllers should recognize that a resource is suspended or unsuspended from the presence of a special metadata key -- this key can be added, removed or changed without patching the object and rolling the generation number.
+
+#### Suspend During Windows
+
+A cluster manager wishes to set periodic windows during which deploys are held back (eg for stability, release control, maintenance, weekends, etc).  Gitops synchronization happens naturally outside of these times.
+
+The cluster manager should be able to define these windows on each resource using a well-known format applied as a metadata key.
+
+#### Seeing Suspend State
+
+Users should be able to see the effective suspend state of the resource with a `flux get` command.  The display should mirror what the controllers interpret the suspend state to be.  This story is included to capture current functionality that should be preserved.
+
+### Alternatives
+
+#### The Gate
+
+The [gate](https://github.com/fluxcd/flux2/pull/3158) proposes a new paired resource and controller to effect a paused reconciliation of selected resources (eg `GitRepository` or `HelmRelease`).
+
+#### More `.spec`
+
+The existing `.spec.suspend` could be expanded with fields for the above semantics.  This would drive more generation number changes and would require a change to the apis.
+
+
+## Design Details
+
+Implementing this RFC would involve resource object status, controllers, cli and metrics.
+
+This feature would create an alternate path to suspending an object and would not violate the current apis.
+
+### Resource Object Status
```

Comment:
It should not be reflected in the Status, as the object would not be reconciled. Instead, the object should have a `IsSuspended()` method which would look at both the `.spec.suspend` and the set annotation.